sdk/ai/azure-ai-projects/samples/agents/sample_agents_image_input_url.py

```diff
@@ -0,0 +1,87 @@
+# pylint: disable=line-too-long,useless-suppression
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+"""
+DESCRIPTION:
+    This sample demonstrates how to use basic agent operations using image url input for the
+    the Azure Agents service using a synchronous client.
+
+USAGE:
+    python sample_agents_image_input_url.py
+
+    Before running the sample:
+
+    pip install azure-ai-projects azure-identity
+
+    Set these environment variables with your own values:
+    1) PROJECT_CONNECTION_STRING - The project connection string, as found in the overview page of your
+       Azure AI Foundry project.
+    2) MODEL_DEPLOYMENT_NAME - The deployment name of the AI model, as found under the "Name" column in 
+       the "Models + endpoints" tab in your Azure AI Foundry project.
+"""
+
+import os, time
+from typing import List
+from azure.ai.projects import AIProjectClient
+from azure.identity import DefaultAzureCredential
+from azure.ai.projects.models import (
+    MessageTextContent,
+    MessageInputContentBlock,
+    MessageImageUrlParam,
+    MessageInputTextBlock,
+    MessageInputImageUrlBlock,
+)
+
+
+project_client = AIProjectClient.from_connection_string(
+    credential=DefaultAzureCredential(),
+    conn_str=os.environ["PROJECT_CONNECTION_STRING"],
+)
+
+with project_client:
+
+    agent = project_client.agents.create_agent(
+        model=os.environ["MODEL_DEPLOYMENT_NAME"],
+        name="my-assistant",
+        instructions="You are helpful assistant",
+    )
+    print(f"Created agent, agent ID: {agent.id}")
+
+    thread = project_client.agents.create_thread()
+    print(f"Created thread, thread ID: {thread.id}")
+
+    image_url = "https://upload.wikimedia.org/wikipedia/commons/thumb/d/dd/Gfp-wisconsin-madison-the-nature-boardwalk.jpg/2560px-Gfp-wisconsin-madison-the-nature-boardwalk.jpg"
+    input_message = "Hello, what is in the image ?"
+    url_param = MessageImageUrlParam(url=image_url, detail="high")
+    content_blocks: List[MessageInputContentBlock] = [
+        MessageInputTextBlock(text=input_message),
+        MessageInputImageUrlBlock(image_url=url_param),
+    ]
+    message = project_client.agents.create_message(thread_id=thread.id, role="user", content=content_blocks)
+    print(f"Created message, message ID: {message.id}")
+
+    run = project_client.agents.create_run(thread_id=thread.id, agent_id=agent.id)
+
+    # Poll the run as long as run status is queued or in progress
+    while run.status in ["queued", "in_progress", "requires_action"]:
+        # Wait for a second
+        time.sleep(1)
+        run = project_client.agents.get_run(thread_id=thread.id, run_id=run.id)
+        print(f"Run status: {run.status}")
+
+    project_client.agents.delete_agent(agent.id)
+    print("Deleted agent")
+
+    messages = project_client.agents.list_messages(thread_id=thread.id)
```

Comment:
Let us do `project_client.agents.list_messages(thread_id=thread.id, order=ListSortOrger.ASCENDING)` to avoid reversing of message list.